src/services/selectDistrictInteractions.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  AbstractInteraction,
+  EventType,
+  GeoJSONLayer,
+  ModificationKeyType,
+} from '@vcmap/core'
+import type { InteractionEvent } from '@vcmap/core'
+import type { RennesApp } from './RennesApp'
+import { apiEnedisDistrictService } from '@/services/api-enedis-district'
+import { useDistrictStore } from '@/stores/districtInformations'
+import { RENNES_LAYER } from '@/stores/layers'
+import Feature from 'ol/Feature'
+import { Point } from 'ol/geom'
+import { updateDistrictPointCoordinates } from './AboveMapService'
+import { Fill, Stroke, Style } from 'ol/style'
+
+const selectedDistrict = new Style({
+  fill: new Fill({
+    color: 'rgba(0,0,0,0.4)',
+  }),
+  stroke: new Stroke({
+    color: '#FFFFFF',
+    width: 2,
+  }),
+})
+
+///// The highlight function doesn't accept an array of styles /////
+
+// export function districtLineStyle() {
+//   const districtStyle: Style[] = []
+//   const firstLine = new Style({
+//     fill: new Fill({
+//       color: 'rgba(0,0,0,0.4)',
+//     }),
+//     stroke: new Stroke({
+//       color: '#FFFFFF',
+//       width: 4,
+
+//     }),
+//     zIndex: 0,
+//   })
+
+//   const secondLine = new Style({
+//     stroke: new Stroke({
+//       color: '#000000',
+//       width: 2,
+//     }),
+//     zIndex: 1,
+//   })
+
+//   districtStyle.push(firstLine)
+//   districtStyle.push(secondLine)
+//   return districtStyle
+// }
+
+class SelectDistrictInteraction extends AbstractInteraction {
+  _rennesApp: RennesApp
+
+  constructor(rennesApp: RennesApp) {
+    super(EventType.CLICK, ModificationKeyType.NONE)
+
+    this._rennesApp = rennesApp
+  }
+
+  async gettingDistrictDatas(codeIris: number) {
+    const districtStore = useDistrictStore()
+    const districtDatas = await apiEnedisDistrictService.getDistrictDatas(
+      codeIris
+    )
+    districtStore.setDistrictInformations(
+      codeIris,
+      districtDatas.irisName,
+      districtDatas.totalConsumption,
+      districtDatas.totalProduction,
+      districtDatas.totalPhotovoltaicSites
+    )
+  }
+
+  _unhighlight() {
+    const irisLayer: GeoJSONLayer = this._rennesApp.layers.getByKey(
+      RENNES_LAYER.iris
+    ) as GeoJSONLayer
+    irisLayer.featureVisibility.clearHighlighting()
+  }
+
+  _highlight(featureId: string | number) {
+    const irisLayer: GeoJSONLayer = this._rennesApp.layers.getByKey(
```

Comment:
You can use await this._rennesApp.getLayerByKey(RENNES_LAYER.iris) 
instead of 
= this._rennesApp.layers.getByKey(
      RENNES_LAYER.iris
    ) as GeoJSONLayer